src/public/modules/forms/admin/controllers/edit-logic-modal.client.controller.js

```diff
@@ -263,19 +268,40 @@ function EditLogicModalController(
 
     if (isNew) {
       vm.formLogics.push(vm.logic)
+      updateLogic({ form_logics: vm.formLogics }).then((error) => {
+        if (!error) {
+          $uibModalInstance.close()
+        }
+      })
       // Not new, and logic index is provided
     } else if (logicIndex !== -1) {
-      vm.formLogics[logicIndex] = vm.logic
+      vm.updateExistingLogic(logicIndex, vm.logic).then((error) => {
+        if (!error) {
+          $uibModalInstance.close()
+        }
+      })
```

Comment:
should we change this to be within `updateExistingLogic` as i'm unsure if an error would actually be thrown here (as it is caught within the `updateExistingLogic` portion)

Reply:
thanks for suggestion, shifted